### PR TITLE
s/input: add SDL controller support

### LINKED ISSUE #278: Feature request: Controller support
Greetings.
Would it be possible to update the xinput control configuration?
TR3 sidesteps are only available with keyboard.
If I use any xinput controller, left sidesteps with the left trigger does not work.
And the rest of the buttons are set as in The Angel of Darkness, not  as in the original PSX version.

### PR DESCRIPTION
**Pull request by rr-**
Resolves #278

#### Checklist

- [x] I have read the [coding conventions](https://github.com/rr-/Tomb1Main/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

I kept Ozi's decision to switch the action and the jump button (I'm a bit conflicted whether to keep it).
I switched Draw Guns with Look to match PS1 controller layout.
I changed D-pad's function to movement. This was dictated by seeing people map D-pad for movement on streams, and Lara herself instructing the player to use the D-pad to move her around.

Tested on my DualShock 4 controller. I have no other devices. I am not a controller player so I'm clueless whether these feel OK.

I want the game to offer sane defaults. Customization comes way later (or perhaps never, or via third party software).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -232,20 +232,21 @@ Not all options are turned on by default. Refer to `Tomb1Main.json5` for details
...
+    | B           | Circle             | Roll/deselect   |
+    | X           | Square             | Action/select   |
+    | Y           | Triangle           | Look/select     |
+    | RB          | R1                 | Walk            |
+    | LB          | L1                 | Draw weapons    |
+    | D-pad       | D-pad              | Movement        |
+    | Back        | Share/Touchpad     | Inventory       |
+    | Start       | Options            | Pause           |
+    | Right Stick | Right stick        | Camera movement |
+    | R3          | R3                 | Reset camera    |
```
Comment (walkawayy) [inline README.md:251]: Should we match the PS1 controller layout? (I didn't update the Xbox column)

    | Xbox button | DualShock button   | Action          |
    | ----------- | ------------------ | --------------- |
    | Left Stick  | Left stick         | Movement        |
    | A           | Square             | Jump            |
    | B           | Circle             | Roll/deselect   |
    | X           | X                  | Action/select   |
    | LB          | Triangle           | Draw weapons    |
    | Y           | L1                 | Look            |
    | LT          | L2                 | Sidestep left   |
    | RB          | R1                 | Walk            |
    | RT          | R2                 | Sidestep right  |
    | D-pad       | D-pad              | Movement        |
    | Back        | Share/Touchpad     | Inventory       |
    | Start       | Options            | Pause           |
    | Right Stick | Right stick        | Camera movement |
    | R3          | R3                 | Reset camera    |

This could also be helped with #564 to allow proper custom control schemes.

Comment (rr-) [reply README.md:251]: Can you also provide which buttons act like select and which deselect?

Comment (walkawayy) [reply README.md:251]: I updated the table with sidesteps too. But after checking out #545 I did a little digging.

TLDR: The select / deselect button has changed a bunch depending on console, region, and game. Since the PS5 release, X is select and circle is deselect in all regions. [(source)](https://www.eurogamer.net/in-a-historic-move-playstation-5-swaps-x-and-circle-button-use-in-japan) I vote we go with that. Custom controls potentially in the future will allow that to be changed.

Long story:
X was always select (for non Asian regions). However, for deselect, it seems different buttons were used. Back in the PS1 days, I remember triangle being the primary deselect button for games. For TR1 though, they used square as deselect for some reason. TR2+ changed deselect to be triangle which matched most games for the PS1-PS2 era.

Since the PS3, X has been select and circle has been deselect. In Asian regions this was reversed, and X was deselect and circle was select. Since the PS5, there is now an all region standard: X is select and circle is deselect.

**Review by oziphantom — APPROVED** [review]
Looks fine, hard to tell exactly how SDL will handle all the mappings across different controllers, the documentation is vague sadly.

**Comment by oziphantom** [comment]
as for switching the xbox buttons its 50/50. I'm not a person who played the original game before T1M, so I have no connection to the "playstation layout", so I made the controls in a sense that makes sense for an "Xbox controller" with xbox conventions. Rather than "I have an Xbox but I think I'm holding a PS1 controller". There is no real answer, but I guess this way it gives people a choice of "classic" and "more modern".

**Comment by rr-** [comment]
Let's go with PS1 era layout walkawayy composed, release as 2.9 to get the universal controller support out there, and then possibly follow up with some sort of customization if enough people want it.

**Comment by walkawayy** [comment]
Ok that sounds like a plan. Hopefully at some point we can get custom controls working properly. Then people are free to customize to their liking. We tested on PS4 and PS5 controllers, but I don't have an Xbox controller to test. When we release 2.9, I guess we will hear some feedback if Xbox controllers have problems lol.

File: src/specific/s_input.c [hunk 2]
```diff
@@ -351,61 +357,134 @@ static bool S_Input_Key(INPUT_ROLE role)
...
+        linput.action       |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_A);
+        linput.select       |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_A);
+        linput.roll         |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_B);
+        linput.deselect     |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_B);
+        linput.jump         |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_X);
+        linput.draw         |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_Y);
+        linput.look         |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_LEFTSHOULDER);
+        linput.slow         |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_RIGHTSHOULDER);
+        linput.option       |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_BACK);
+        linput.deselect     |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_TOUCHPAD);
```
Comment (walkawayy) [inline src/specific/s_input.c:483]: Is deselect supposed to be used twice here? Set to TOUCHPAD and B button.

Comment (rr-) [reply src/specific/s_input.c:483]: Hmm I kept it because I felt like it but I can remove it.

Comment (walkawayy) [reply src/specific/s_input.c:483]: Oh that's fine. Just wanted to make sure it wasn't an oversight. Makes sense to be able to close the inventory in game with the same button you opened it with.

Comment (rr-) [reply src/specific/s_input.c:483]: On that note maybe I should add this behavior to the pause and the alt inventory button too?

Comment (walkawayy) [reply src/specific/s_input.c:483]: Sure. Makes sense.

Comment (rr-) [reply src/specific/s_input.c:483]: Addressed

**Review by walkawayy — APPROVED** [review]
LGTM
